### PR TITLE
Add table of users indices #138

### LINKED ISSUE #138: Add table of user's indices to profile page

### PR DESCRIPTION
**Pull request by grimm**
Do we want to have these actions in the user's profile?  It will cause issues on how the update and create actions return.

Fixes #138

### CODE REVIEW

**Comment by teknofire** [comment]
no, it should just take you to the show page for the index, that will have the necessary buttons to perform those actions

**Comment by teknofire** [comment]
which is part of what I'm working on now in #142

**Comment by teknofire** [comment]
Probably should have the text on the link be something like `View AWRVI Dashboard`

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

File: app/models/index.rb [hunk 1]
```diff
@@ -20,6 +20,8 @@ def publish!(at = Time.zone.now)
     update(published_at: at)
   end
 
+  scope :user_indices, ->(user) { where(user_id: user) }
```
Comment (teknofire) [inline app/models/index.rb]: you should be able to do the same thing using `current_user.indices` relation

File: app/views/manage/users/show.html.haml [hunk 2]
```diff
@@ -2,4 +2,20 @@
...
\ No newline at end of file
+
+%h3 Your Indices:
+%table.table.table-condensed.table-striped
+  %thead
+    %tr
+      %th Community
+      %th Show Index
+  %tbody
+    - Index.user_indices(current_user).each do |index|
```
Comment (teknofire) [inline app/views/manage/users/show.html.haml]: Better method for this is to pull these out into an instance variable in the manager/users controller

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

**Comment by teknofire** [comment]
We should add some information to the list,

Like when it was last updated, how complete is it, if it's finalized or not

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

File: app/views/manage/users/show.html.haml [hunk 3]
```diff
@@ -2,4 +2,30 @@
...
+      %th.col-sm-2 View AWRVI Dashboard
+  %tbody
+    - @indices.each do |index|
+      %tr
+        %td
+          %b= index.community.name
+        %td
+          #{number_with_precision(index.vulnerability_index, precision: 2)}
+          - unless index.published_at.nil?
+            .label.label-warning Published
```
Comment (smacfarlane) [inline app/views/manage/users/show.html.haml]: I think we want to use label-primary to be consistent with the styling on the community show page.

File: app/views/manage/users/show.html.haml [hunk 4]
```diff
@@ -2,4 +2,30 @@
...
+        %td
+          #{time_ago_in_words index.updated_at} ago
+        %td
+          .progress
+            - percentage = index.vulnerability_index * 100
```
Comment (smacfarlane) [inline app/views/manage/users/show.html.haml]: The label says progress, but what's being displayed is the vulnerability index.  I think completeness_percentage is the method we probably want to be using.

File: app/views/manage/users/show.html.haml (hunk 3 above)
Comment (grimm) [inline app/views/manage/users/show.html.haml]: I just used this because the publish button has the same color.

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

File: app/views/manage/users/show.html.haml [hunk 5]
```diff
@@ -2,4 +2,30 @@
...
+        %td
+          #{number_with_precision(index.vulnerability_index, precision: 2)}
+          - unless index.published_at.nil?
+            .label.label-primary Published
+        %td
+          #{time_ago_in_words index.updated_at} ago
+        %td
+          .progress
+            - percentage = index.completeness_percentage
+            .progress-bar{style: "width: #{percentage}%", class: progress_bar_for_percentage(percentage)}
```
Comment (teknofire) [inline app/views/manage/users/show.html.haml]: Do we want to use `progress_bar_for_percentage` for the completeness?  That's really intended for the vulnerability index coloring handling

Comment (teknofire) [inline app/views/manage/users/show.html.haml]: If we are being consistent with how we are handling this for the main index dashboards anway

https://github.com/gina-alaska/awrvi/blob/master/app/views/indices/_index.html.haml#L7

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

**Comment by teknofire** [comment]
Looks good to me 👍

**Comment by smacfarlane** [comment]
:icecream:  :+1:

**Pull request merged**